iotilecore/test/test_hw/test_virtualadapter.py

```diff
@@ -156,3 +167,11 @@ def test_realtime_tracing(tracer_hw):
 
     wrong_data = [x for x in words if (x != 'hello' and x != 'goodbye' and x != '')]
     assert len(wrong_data) == 0
+
+def test_virtual_scan(realtime_scan_hw):
+    """Make sure we can scan for virtual devices and connect directly without connect_direct
+    """
+    devices = report_hw.scan()
```

Comment:
You can just push a fix to the same branch and the PR will update and the CI will run again.  Other than that it looks good.